bin.src/thresholdFits.py

```diff
@@ -0,0 +1,271 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.    See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+import os
+import argparse
+from glob import glob
+import astropy.io.fits as fits
+
+
+def expand_files(filename_list=None):
+    """Expand a set of filenames and globs, returning a list of filenames
+
+    Parameters
+    ------------
+    filename_list:  list
+        A list of files and glob patterns
+
+    Returns:
+    -----------
+    expandedList : list
+        An expanded list of files that exist and files that fit the glob pattern
+    """
+    expanded_list = []
+    for globPattern in filename_list:
+        files = glob(globPattern)
+
+        if not files:
+            print("%s doesn't match any file" % globPattern)
+            continue
+
+        expanded_list.extend(files)
+
+    return expanded_list
+
+
+def get_size(filepath=None):
+    """
+    Convenience function to return the filesize.
+
+    Parameters:
+    -----------
+    filepath : str
+        Path to the file to check
+
+    Returns:
+    --------
+    size : str
+        Human-readable filesize (Kilobytes or Megabytes)
+    """
+    size_bytes = os.path.getsize(filepath)
+    if size_bytes < (1024 * 1024):
+        size = str(round(size_bytes / 1024)) + "K"
+    else:
+        size = str(round((size_bytes / (1024 * 1024)), 1)) + "M"
+    return size
+
+
+def threshold_file(
+    fits_file=None,
+    threshold=None,
+    not_binary=False,
+    subtract_threshold=False,
+    keep_original_value=True,
+    out_dir=None,
+):
+    """
+    Threshold a FITS file, by changing the value of pixels for
+    each .data element. By default, pixels below the threshold
+    are set to 0, and those above to 1. The headers are kept
+    unchanged.
+
+    Parameters:
+    -----------
+    fits_file : str
+        Path to the FITS file to threshold
+    threshold : float, optional
+        A value of threshold to apply
+    not_binary: bool, optional
+        If set to True, do non-binary thresholding.
+    subtract_threshold: bool, optional
+        If set to True, set pixels below threshold to 0,
+        from those above, subtract the threshold
+    keep_original_value: bool, optional
+        If set to True, set pixels below threshold to 0,
+        keep the value of those above threshold
+    out_dir : str
+        Path to output directory
+    """
+    print("\nThresholding %s" % fits_file)
+
+    hdul = fits.open(fits_file)
```

Comment:
Did you close the file after the opening?

Reply:
Fixed adding `hdul.close()`